openedx/core/djangoapps/xblock/rest_api/views.py

```diff
@@ -7,27 +7,30 @@
 
 from corsheaders.signals import check_request_enabled
 from django.contrib.auth import get_user_model
+from django.http import Http404
 from django.views.decorators.clickjacking import xframe_options_exempt
 from django.views.decorators.csrf import csrf_exempt
 from rest_framework import permissions
 from rest_framework.decorators import api_view, permission_classes  # lint-amnesty, pylint: disable=unused-import
-from rest_framework.exceptions import PermissionDenied, AuthenticationFailed
+from rest_framework.exceptions import PermissionDenied, AuthenticationFailed, NotFound
 from rest_framework.response import Response
 from xblock.django.request import DjangoWebobRequest, webob_to_django_response
 
+from opaque_keys import InvalidKeyError
 from opaque_keys.edx.keys import UsageKey
 from openedx.core.lib.api.view_utils import view_auth_classes
 from ..api import (
     get_block_metadata,
     get_handler_url as _get_handler_url,
     load_block,
     render_block_view as _render_block_view,
-
 )
 from ..utils import validate_secure_token_for_xblock_handler
 
 User = get_user_model()
 
+not_found_invalid_exception = NotFound("Invalid XBlock key")
```

Comment:
Usually we define classes by subclassing. Haven't seen the pattern of constructing an instance. And even the former is useful for Python APIs and not web ones where the exception class will get converted into an HTTP response. 🤔 

* https://github.com/django/django/blob/main/django/http/response.py#L517-L536
* https://github.com/edx/edx-platform/blob/master/openedx/core/lib/blockstore_api/exceptions.py#L6